src/nvim/window.c

```diff
@@ -2301,20 +2305,22 @@ static win_T *frame2win(frame_T *frp)
   return frp->fr_win;
 }
 
-/*
- * Return TRUE if frame "frp" contains window "wp".
- */
-static int frame_has_win(frame_T *frp, win_T *wp)
+/// Return true if the frame "frp" contains the window "wp".
+///
+/// @param frp frame
+/// @param wp  window
+static bool frame_has_win(frame_T *frp, win_T *wp)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
```

Comment:
Though I guess this is FUNC_ATTR_NONNULL_ALL, but `wp` is not dereferenced and can be NULL thus.